### ISSUE TITLE
Fast: Documentation: Screen reader fails announcing about the group name associated with suggestions available while navigating along the suggestions.

### ISSUE THREAD

**Issue by msftedad** [facebook/docusaurus#9175]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

Screen reader fails announcing about the group name associated with suggestions available while navigating along the suggestions.
User is not able to know the group name associated with suggestions available.
Screen reader should announce about the group associated with suggestions available while navigating along the suggestions. i.e., When user enter the suggestions group, screen reader should announce as '<Group Name>and <suggestion name> along <position set>.
User with visual impairment who rely on screen reader will face difficulties if associated group name is not announced by screen reader.

### Steps to reproduce

1. Open https://docusaurus.io/docs URL.
2. Navigate to 'Search' button using tab key and invoke it using enter key.
3. Search for any keyword i.e. 'test' and navigate to suggestions available using tab key. 
4. Verify and observe whether Screen Reader is announcing about the group name associated with suggestions available while navigating along the suggestions using tab key or not.

Attachment: 
[Associated label is not announced .webm](https://github.com/facebook/docusaurus/assets/93735775/919bd82b-e0fb-4766-8cf2-5e6282a5aa82)

### Expected behavior

Screen reader should announce about the group associated with suggestions available while navigating along the suggestions. i.e., When user enter the suggestions group, screen reader should announce as '<Group Name>and <suggestion name> along <position set>.

### Actual behavior

Screen reader fails announcing about the group name associated with suggestions available while navigating along the suggestions.

i.e., When user navigates to any suggestion, screen reader announces as '<Suggestion label><position set>'.

### Your environment

Public source code:
Public site URL: https://www.fast.design/
Docusaurus version used:
OS: Windows 11, Version: 22H2, Build: 22621.1992  
Microsoft Edge Version: 114.0.1823.86 (Official build) (64-bit)
Screen Reader used: NVDA
[Associated label is not announced .webm](https://github.com/facebook/docusaurus/assets/93735775/7f23c97a-71e6-47f2-a39f-6895d2a1d5b2)

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by pinakipb2** [comment]
@msftedad Can you please check and confirm if the screen reading is correct for first 3 element in the list under "Community Contribution".

Download the full folder, and run only html file to check.

Html file to check : https://drive.google.com/drive/folders/1UC9AnsSzqbFgNOeUV3R9x4OSyKBwUaAD?usp=sharing

**Comment by pinakipb2** [comment]
Hi @msftedad, ```but not reading as list 1 of 4 and 4 of 4 list count``` you have mentioned it neither in the issue nor in the explanation video. Can you explain the whole screen reader text once more along with navigation for easy understanding.
Thanks.

**Comment by msftedad** [comment]
@pinakipb2, The list count is read in the URL provided by us so we have not mentioned that point but not reading in the html code provided by you.
Thanks.

**Comment by pinakipb2** [comment]
> @pinakipb2, The list count is read in the URL provided by us so we have not mentioned that point but not reading in the html code provided by you. Thanks. <img alt="Screenshot (116)" width="916" src="https://user-images.githubusercontent.com/93735775/262945071-bfbc4c47-24c9-4cb6-bb61-794cc00b1e95.png">

Hi @msftedad , you should also download the other folder in the drive link, not only the html. The paths must be same as in the drive. Then open the html and check. Thanks

**Comment by pinakipb2** [comment]
> > @pinakipb2, The list count is read in the URL provided by us so we have not mentioned that point but not reading in the html code provided by you. Thanks. <img alt="Screenshot (116)" width="916" src="https://user-images.githubusercontent.com/93735775/262945071-bfbc4c47-24c9-4cb6-bb61-794cc00b1e95.png">
> 
> Hi @msftedad , you should also download the other folder in the drive link, not only the html. The paths must be same as in the drive. Then open the html and check. Also can you share video where it is read. Or share where it is read in the URL provided. As far as I can the video URL that you have attached in the issue does not say anything like 1 of 4, it just says "link test". Kindly recheck or provide a new video that would be really helpful.
> 
> Thanks

@msftedad Please go through it.

**Comment by pinakipb2** [comment]
> > @pinakipb2, The list count is read in the URL provided by us so we have not mentioned that point but not reading in the html code provided by you. Thanks. <img alt="Screenshot (116)" width="916" src="https://user-images.githubusercontent.com/93735775/262945071-bfbc4c47-24c9-4cb6-bb61-794cc00b1e95.png">
> 
> Hi @msftedad , you should also download the other folder in the drive link, not only the html. The paths must be same as in the drive. Then open the html and check. Also can you share video where it is read. Or share where it is read in the URL provided. As far as I can the video URL that you have attached in the issue does not say anything like 1 of 4, it just says "link test". Kindly recheck or provide a new video that would be really helpful.
> 
> Thanks

Hi @msftedad, please go through it.

**Comment by msftedad** [comment]
Hi @pinakipb2, Now the screen reader is announcing the list items with associated heading. Thanks.

**Comment by pinakipb2** [comment]
Hi @msftedad, I can confirm that this issue is not from docusaurus side but from ("https://docsearch.algolia.com/") their end as the modal is rendered by them we can change the look and style for it but unfortunately not the rendered html.
Thanks

**Comment by pinakipb2** [comment]
Hi @msftedad, I can confirm that this issue is not from docusaurus side but from ("https://docsearch.algolia.com/") their end as the modal is rendered by them we can change the look and style for it but unfortunately not the rendered html.
Thanks

**Comment by pinakipb2** [comment]
@slorber @msftedad feel free to close this issue. 
![image](https://github.com/facebook/docusaurus/assets/48859773/4eb58121-0c40-49dc-a494-e06d179c4b8e)

**Comment by pinakipb2** [comment]
@slorber @msftedad feel free to close this issue. 
![image](https://github.com/facebook/docusaurus/assets/48859773/4eb58121-0c40-49dc-a494-e06d179c4b8e)

**Comment by slorber** [comment]
Hey

It's hard for me to understand this issue.  
But I think by "suggestions", you meant "Algolia DocSearch autocomplete results".

I'm not sure what's the accessibility issue here, but if there is one, it should rather be reported at https://github.com/algolia/docsearch, cc @shortcuts
